lib/dependent.js

```diff
@@ -0,0 +1,35 @@
+const _ = require('lodash')
+
+/**
+ * process the PR based on dependent settings
+ *
+ * @return
+ *  JSON object with the properties mergeable and the description if
+ *  `mergeable` is false
+ */
+module.exports = async (pr, context, settings) => {
+  let dependentSettings = settings.dependent
+  if (!dependentSettings) {
+    return {
+      mergeable: true,
+      description: 'Okay To Merge'
+    }
+  }
+
+  let dependentFiles = settings.dependent.files
+  let message = settings.dependent.message ||
+    'One of the following file is modified, all the other files in the list must be modified as well ' + dependentFiles.map(name => '\n  -' + name)
```

Comment:
Probably want a colon at the end:
"One of the following file is modified, all the other files in the list must be modified as well:"

I believe in markdown for bullet points to work you need a space in between the dash and the sentence:
`dependentFiles.map(name => '\n  - ' + name)`